addons/popochiu/engine/objects/gui/components/dialog_text/dialog_caption/dialog_caption.gd

```diff
@@ -4,14 +4,40 @@ extends PopochiuDialogText
 #region Private ####################################################################################
 func _modify_size(msg: String, _target_position: Vector2) -> void:
 	var _size := await _calculate_size(msg)
-	
+
 	# Define size and position (before calculating overflow)
+	rich_text_label.size.x = _size.x
 	rich_text_label.size.y = _size.y
+	rich_text_label.position.x = (get_viewport_rect().size.x/2) - (_size.x /2)
 	rich_text_label.position.y = get_meta(DFLT_POSITION).y - (_size.y - get_meta(DFLT_SIZE).y)
 
 
 func _append_text(msg: String, props: Dictionary) -> void:
+	msg = _correct_line_breaks(msg)
 	rich_text_label.text = "[center][color=%s]%s[/color][/center]" % [props.color.to_html(), msg]
 
 
+## Appends text for the dialog caption
+## Ensures that where a printing a word would see it wrap to the next line that the newline
+## is forced into the text. Without this the tween in dialog_text.gd would print part of the word
+## until it runs out of space, then erase the part word and rewrite it on the next line which looks
+## messy.
+func _correct_line_breaks(msg: String) -> String:
```

Comment:
I think `_fix_line_breaks` is a better name for this function. "Correct" sounds weird.